packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackset-action.ts

```diff
@@ -0,0 +1,691 @@
+   */
+  readonly deploymentModel?: StackSetDeploymentModel;
+
+  /**
+   * The template parameters for your stack set
+   *
+   * These parameters are shared between all instances of the stack set.
+   *
+   * @default - No parameters will be used.
+   */
+  readonly parameters?: StackSetParameters;
+
+  /**
+   * Indicates that the template can create and update resources, depending on the types of resources in the template.
+   *
+   * You must use this property if you have IAM resources in your stack template or you create a stack directly from a template containing macros.
+   *
+   * @default - The StackSet will have no IAM capabilities.
+   */
+  readonly cfnCapabilities?: cdk.CfnCapabilities[];
+
+  /**
+   * The percentage of accounts per Region for which this stack operation can fail before AWS CloudFormation stops the operation in that Region. If
+   * the operation is stopped in a Region, AWS CloudFormation doesn't attempt the operation in subsequent Regions. When calculating the number
+   * of accounts based on the specified percentage, AWS CloudFormation rounds down to the next whole number.
+   *
+   * @default 0
+   */
+  readonly failureTolerancePercentage?: number;
+
+  /**
+   * The maximum percentage of accounts in which to perform this operation at one time. When calculating the number of accounts based on the specified
+   * percentage, AWS CloudFormation rounds down to the next whole number. If rounding down would result in zero, AWS CloudFormation sets the number as
+   * one instead. Although you use this setting to specify the maximum, for large deployments the actual number of accounts acted upon concurrently
+   * may be lower due to service throttling.
+   *
+   * @default 1
+   */
+  readonly maxConcurrentPercentage?: number;
+
+  /**
+   * The AWS region the given Action resides in.
+   * Note that a cross-region Pipeline requires replication buckets to function correctly.
+   * You can provide their names with the {@link PipelineProps#crossRegionReplicationBuckets} property.
+   * If you don't, the CodePipeline Construct will create new Stacks in your CDK app containing those buckets,
+   * that you will need to `cdk deploy` before deploying the main, Pipeline-containing Stack.
+   *
+   * @default - the Action resides in the same region as the Pipeline
+   */
+  readonly region?: string;
+}
+
+/**
+ * Where Stack Instances will be created from the StackSet
+ */
+export abstract class StackInstances {
+  /**
+   * Create stack instances in a literal set of accounts or organizational units, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromList(targets: string[], regions: string[]): StackInstances {
```

Comment:
I feel like this is too generic. Because we've bundled accounts and OUs together, we have to use generic terms like `fromList()` and `targets`, instead of more specific ones.

How about splitting them up into `inAccountsAndRegions(accounts: string[], regions: string[])` and `inOrganizationalUnitsAndRegions(oUs: string[], regions: string[])`? This way, we could actually validate that, if you're using `inOrganizationalUnitsAndRegions()`, you're also using `StackSetDeploymentModel.organizations()`, and throw if that isn't true?

Reply:
But wait:

* We are inventing a synthetic distinction
* And then slapping you on the wrist if you don't use the correct variant of the two things we invented

Not to mention that on `StackInstancesAction` we won't be able to tell if the StackSet has been configured for use with Organizations or use with plain Accounts, so we won't be able to do the validation there.

I had it modeled around OUs vs Accounts at first... until I implemented `StackInstancesAction` and found out that that distinction goes right out the window there.

I don't necessarily mind adding the two factory functions (so we will have three: `inAccountsAndRegions`, `inOrganizationUnitsAndRegions`, `inArtifactTargets` ... ?) but we're going to be very clear that they have no semantic meaning and are just there to make your code look nicer.

I actually don't like the `AndRegions` part of that either. Way too long, no? `inAccounts`, `inOrganizationalUnits`.

Reply:
I still think there's value here. The `fromList()` just looks so ugly 😕.

I'm fine with `inAccounts()`, `inOrganizationalUnits()`.